### PR TITLE
[TextFields] Themer Example

### PR DESCRIPTION
**Pull request by ianegordon**
Add an example that themes an Outlined and Filled TextField.

### CODE REVIEW

File: components/TextFields/examples/TextFieldSemanticColorThemer.swift [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+                      NSLayoutConstraint(item: textfieldAlternative,
+                                         attribute: .bottom,
+                                         relatedBy: .equal,
+                                         toItem: scrollView,
+                                         attribute: .bottomMargin,
+                                         multiplier: 1,
+                                         constant: -20)]
+    }
+    #else
+    constraints += [NSLayoutConstraint(item: textfieldStandard,
```
Comment (yarneo) [inline components/TextFields/examples/TextFieldSemanticColorThemer.swift:139]: indentation

File: components/TextFields/examples/TextFieldSemanticColorThemer.swift [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+    attribute: .top,
+    relatedBy: .equal,
+    toItem: scrollView,
+    attribute: .top,
+    multiplier: 1,
+    constant: 20),
+    NSLayoutConstraint(item: textfieldAlternative,
```
Comment (yarneo) [inline components/TextFields/examples/TextFieldSemanticColorThemer.swift]: indentation

File: components/TextFields/examples/TextFieldSemanticColorThemer.swift [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+    #endif
+
+    NSLayoutConstraint.activate(constraints)
+  }
+
+  func setupScrollView() {
+    view.addSubview(scrollView)
+    scrollView.translatesAutoresizingMaskIntoConstraints = false
+
+    NSLayoutConstraint.activate(NSLayoutConstraint.constraints(
```
Comment (yarneo) [inline components/TextFields/examples/TextFieldSemanticColorThemer.swift]: please stay consistent with the indentation of the one below

File: components/TextFields/examples/TextFieldSemanticColorThemer.swift [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+  }
+
+}
+
+extension TextFieldSemanticColorThemer: UITextFieldDelegate {
+  func textField(_ textField: UITextField,
+                 shouldChangeCharactersIn range: NSRange,
+                 replacementString string: String) -> Bool {
+    guard let _ = textField.text else {
+      return true
```
Comment (yarneo) [inline components/TextFields/examples/TextFieldSemanticColorThemer.swift]: should this be false?

File: components/TextFields/examples/TextFieldSemanticColorThemer.swift [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+  }
+
+  func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+    return false
+  }
+}
+
+extension TextFieldSemanticColorThemer: UITextViewDelegate {
+  func textViewDidEndEditing(_ textView: UITextView) {
+    print(textView.text)
```
Comment (yarneo) [inline components/TextFields/examples/TextFieldSemanticColorThemer.swift]: do you want to keep this print?

File: components/TextFields/examples/TextFieldSemanticColorThemer.swift [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+
+  @objc func keyboardWillHide(notif: Notification) {
+    scrollView.contentInset = UIEdgeInsets()
+  }
+}
+
+// MARK: - Status Bar Style
+
+extension TextFieldSemanticColorThemer {
+  @objc class func catalogBreadcrumbs() -> [String] {
```
Comment (yarneo) [inline components/TextFields/examples/TextFieldSemanticColorThemer.swift:231]: if you want this part of the catalog app don't forget to add catalogIsPresentable

File: components/TextFields/examples/TextFieldSemanticColorThemer.swift [hunk 7]
```diff
@@ -0,0 +1,259 @@
...
+      return
+    }
+    scrollView.contentInset = UIEdgeInsets(top: 0.0,
+                                           left: 0.0,
+                                           bottom: frame.height,
+                                           right: 0.0)
+  }
+
+  @objc func keyboardWillHide(notif: Notification) {
+    scrollView.contentInset = UIEdgeInsets()
```
Comment (yarneo) [inline components/TextFields/examples/TextFieldSemanticColorThemer.swift]: `= .zero`

**Review by yarneo — DISMISSED** [review]

File: components/TextFields/examples/TextFieldSemanticColorThemer.swift (hunk 6 above)
Comment (ianegordon) [reply components/TextFields/examples/TextFieldSemanticColorThemer.swift:231]: Dragon Only.

File: components/TextFields/examples/TextFieldSemanticColorThemer.swift (hunk 1 above)
Comment (ianegordon) [reply components/TextFields/examples/TextFieldSemanticColorThemer.swift:139]: C&P Formatting fixed.

File: components/TextFields/examples/TextFieldSemanticColorThemer.swift (hunk 2 above)
Comment (ianegordon) [reply components/TextFields/examples/TextFieldSemanticColorThemer.swift]: C&P Formatting fixed.

File: components/TextFields/examples/TextFieldSemanticColorThemer.swift (hunk 3 above)
Comment (ianegordon) [reply components/TextFields/examples/TextFieldSemanticColorThemer.swift]: C&P Formatting fixed.

File: components/TextFields/examples/TextFieldSemanticColorThemer.swift (hunk 4 above)
Comment (ianegordon) [reply components/TextFields/examples/TextFieldSemanticColorThemer.swift]: Nuked.

File: components/TextFields/examples/TextFieldSemanticColorThemer.swift (hunk 5 above)
Comment (ianegordon) [reply components/TextFields/examples/TextFieldSemanticColorThemer.swift]: Nuked.

File: components/TextFields/examples/TextFieldSemanticColorThemer.swift (hunk 7 above)
Comment (ianegordon) [reply components/TextFields/examples/TextFieldSemanticColorThemer.swift]: Done.

**Review by yarneo — APPROVED** [review]

**Comment by jverkoey** [comment]
Is there a technical reason not to have this in MDCCatalog? We generally do want good examples of theming in the main catalog.

**Comment by ianegordon** [comment]
Catalog *ideally* should only have beautiful examples that show off our components.  Dragons is for everything else.  This is definitely on the everything else side of the fence :)

**Pull request merged**